x-pack/plugins/secops/public/components/page/network/network_dns_table/index.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiBadge, EuiFlexGroup, EuiFlexItem, EuiIconTip } from '@elastic/eui';
+import { isEqual } from 'lodash/fp';
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import { ActionCreator } from 'typescript-fsa';
+
+import {
+  Direction,
+  NetworkDnsEdges,
+  NetworkDnsFields,
+  NetworkDnsSortField,
+} from '../../../../graphql/types';
+import { networkActions, networkModel, networkSelectors, State } from '../../../../store';
+import { Criteria, ItemsPerRow, LoadMoreTable } from '../../../load_more_table';
+
+import { getNetworkDnsColumns } from './columns';
+import { IsPtrIncluded } from './is_ptr_included';
+import * as i18n from './translations';
+
+interface OwnProps {
+  data: NetworkDnsEdges[];
+  loading: boolean;
+  hasNextPage: boolean;
+  nextCursor: string;
+  loadMore: (cursor: string) => void;
+  startDate: number;
+  totalCount: number;
+  type: networkModel.NetworkType;
+}
+
+interface NetworkDnsTableReduxProps {
+  limit: number;
+  dnsSortField: NetworkDnsSortField;
+  isPtrIncluded: boolean;
+}
+
+interface NetworkDnsTableDispatchProps {
+  updateDnsSort: ActionCreator<{
+    dnsSortField: NetworkDnsSortField;
+    networkType: networkModel.NetworkType;
+  }>;
+  updateDnsLimit: ActionCreator<{
+    limit: number;
+    networkType: networkModel.NetworkType;
+  }>;
+  updateIsPtrIncluded: ActionCreator<{
+    isPtrIncluded: boolean;
+    networkType: networkModel.NetworkType;
+  }>;
+}
+
+type NetworkDnsTableProps = OwnProps & NetworkDnsTableReduxProps & NetworkDnsTableDispatchProps;
+
+const rowItems: ItemsPerRow[] = [
+  {
+    text: i18n.ROWS_5,
+    numberOfRow: 5,
+  },
+  {
+    text: i18n.ROWS_10,
+    numberOfRow: 10,
+  },
+  {
+    text: i18n.ROWS_20,
+    numberOfRow: 20,
+  },
+  {
+    text: i18n.ROWS_50,
+    numberOfRow: 50,
+  },
+];
+
+const CountBadge = styled(EuiBadge)`
+  margin-left: 5px;
+`;
+
+const Sup = styled.sup`
+  vertical-align: super;
```

Comment:
Don't see many of the super script alignments but I assume this is what we want.